### PR TITLE
PP-7276 Emit email collected event

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT YOU DID
- Emits separate event to ledger when email is patched by frontend. This is so email can be sent to ledger
  even when user cancels payment during authoration operation.
  When a charge is cancelled by user, authorisation operation in progress fails on invalid state transition and no further events related to authorisation are emitted to ledger.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/charge/service/ChargeService.java [hunk 1]
```diff
@@ -321,6 +321,7 @@ private CardDetailsEntity createCardDetailsEntity(PrefilledCardHolderDetails pre
                 .map(chargeEntity -> {
                     if (chargePatchRequest.getPath().equals(ChargesApiResource.EMAIL_KEY)) {
                         chargeEntity.setEmail(sanitize(chargePatchRequest.getValue()));
+                        eventService.emitAndRecordEvent(UserEmailCollected.from(chargeEntity));
```
Comment (sfount) [inline src/main/java/uk/gov/pay/connector/charge/service/ChargeService.java:324]: As this method is `@Transactional`, will the charge still be updated if the emit and record event method throws anything other than `QueueException` (for some reason I can't think of off the top of my head).

Is it worth even further guarding against this failing because the event failed or do we think that will make the exception handling hard to follow?

**Review by stephencdaly — APPROVED** [review]
Looks good to me

File: src/main/java/uk/gov/pay/connector/charge/service/ChargeService.java (hunk 1 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/charge/service/ChargeService.java:324]: For any exceptions transaction will be rolled back. 

Exceptions thrown from downstream code are `ChargeEventNotFoundRuntimeException` or any `runtime exception` (if QueueException can't handle the expected in event service).

`ChargeEventNotFoundRuntimeException` shouldn't happen as it is raised only when charge doesn't CREATED or ENTERING_CARD_DETAILS event.

We can handle all exceptions in EventService instead of only QueueException to be extra safe which can also cover exception handling for `PaymentDetailsEntered` event. (will raise separate PR if we would like go with this)
